### DISCUSSION TITLE
Default nuget details

### DISCUSSION THREAD

**Discussion by Const-me** [microsoft/vcpkg#21830] [Ideas]
When I run `./vcpkg export <single package> --nuget` the output file name defaults to `vcpkg-export-<timestamp>.1.0.0.nupkg`

Instead, it should default to `<package><version>.nupkg`

I've passed package name to the command line argument. Packages have versions in their CONTROL files. You already have good default values for both --nuget-id and --nuget-version parameters.

**Comment by invy** [comment]
1. This makes no sense, as vcpkg doesn't export by default 'single package'. It packs all dependent packages.
2. Automatically putting "version" based on version stored in CONTROL file won't work out of the box, as NuGet are versioned with SemVer and arbitraty libraries have their own versioning strategies.
3. You won't be able to import multiple 'single packages' (see linket PR and issues)

We've implemented single package exporting (including dependencies), which works and deals with most issues: https://github.com/microsoft/vcpkg/compare/master...invym:master

Implementation is based on: #2527

**Comment by Const-me** [comment]
@invy 1. I understand the shared dependency issue, but in my experience, many authors of C and C++ libraries try to minimize their dependencies.

2. A function like `bool isSemanticVersion( string libraryVersion )` is not too hard to write. If the version is not a semantic one, you can default to `1.0` like you’re doing now.

P.S. In our team nuget packages is the only reason why I’m still trying to use vcpkg. At least this way I’m the only person spending time building them and fixing numerous port bugs. If every C++ developer would have to do what I’m doing, I would have banned vcpkg usage in our team long ago, it takes too much time due to usability issues like this and many others you don’t want to fix (like the default 32-bit build target).

**Comment by invy** [comment]
@Const-me you could try our implementation, it works for us, the only reason we didn't file a PR is CLA. The one thing we didn't do is automatically assign version... because of versions of packages (it just didn't fit anywhere for bunch of packages), so for us the version is the part of ID and nuget package version either defaults to 1.0.0 or we can specify it.
